### PR TITLE
Add docs & untested pages workflow

### PR DESCRIPTION
**Pull request by armins**
* Adds jekyll pages documentation from 'docs' branch, cleaned up a bit.
* Workflow to test jekyll output with htmlproofer
* Workflow to build & deploy to GH pages

### CODE REVIEW

**Comment by armins** [comment]
Should we change the workflow condition to run on the feature branch?
Otherwise we will not know, if the build & deploy to GH pages is working before merging.

**Comment by armins** [comment]
Makes 'docs' branch obsolete

File: .github/workflows/jekyll-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# documentation.
+
+# Sample workflow for building and deploying a Jekyll site to GitHub Pages
+name: Deploy Jekyll site to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+    paths: ["docs/**/*.markdown"]
```
Comment (whotwagner) [inline .github/workflows/jekyll-gh-pages.yml:13]: why limiting to ".markdown" only?

**Review by whotwagner — CHANGES_REQUESTED** [review]

File: .github/workflows/jekyll-gh-pages.yml (hunk 1 above)
Comment (armins) [reply .github/workflows/jekyll-gh-pages.yml]: As discussed: change to all in `docs` dir

**Review by whotwagner — APPROVED** [review]

**Pull request merged**